test/unit/libraries/StableMath.t.sol

```diff
@@ -0,0 +1,64 @@
+pragma solidity ^0.8.0;
+
+import "test/__fixtures/BaseTest.sol";
+
+import { StableMath } from "src/libraries/StableMath.sol";
+import { Math } from "@openzeppelin/utils/math/Math.sol";
+
+contract StableMathTest is BaseTest {
+
+    function testMinALessThanMaxA() public
+    {
+        // assert
+        assertLt(StableMath.MIN_A, StableMath.MAX_A);
+    }
+
+    function testComputeLiquidityFromAdjustedBalances_ConvergeEvenWithVeryUnbalancedValues (
+        uint256 aReserve0, uint256 aReserve1, uint256 aN_A
+    ) public
+    {
+        // assume - covers ratios up to 1:1000, which is good enough even in the case of a depeg
+        uint256 lReserve0 = bound(aReserve0, 1e6, type(uint112).max);
+        uint256 lReserve1 = bound(aReserve1, lReserve0 / 1e3, Math.min(type(uint112).max, lReserve0 * 1e3));
+        uint256 lN_A = bound(aN_A, StableMath.MIN_A * StableMath.A_PRECISION, StableMath.MAX_A * StableMath.A_PRECISION);
```

Comment:
okay so here's the thing, having fuzzed a lot of iterative functions, I noticed that the function does not converge on some random values sometimes. On average once every 150 fuzzes

It is not that when the difference between two values goes beyond a certain threshold then it fails. Sometimes relatively close values (within `1e6`) can also fail. On the other hand differences beyond `1e22` can still converge. 

Examples of values that don't converge 

```
4280926068553959580125235477587551
428092606855395958012523547758

17515493530788801010825324439190
1751549353078880101082532454

1001927
103

974523083462972163360456
97452308346297216570

1000000
13

2828486865529418933375394711587169
116901276727300513218678699616


1000000
4874083481260427647068269489

2244282415425521217040721616
1000000
```

I don't know what's this pattern... Every time the ratios are close to some power of 10 it fails? Or when one number is 1e21 larger than the other number?

Reply:
It enters a loop before 100 iterations and stays there forever. Seems cyclical:

![image](https://user-images.githubusercontent.com/11343499/197405321-016b2960-0b7a-416b-9700-7e87e13cd9fa.png)

It appears to have two causes:

1. Sometimes can only get within 2 (could just be an unlucky rounding scenario).
2. Has a maximum accuracy, for instances HurHur2 is accurate within 0.0000000000000029%.

I see a few options:

1. Keep the revert if we assume it exceedingly rare for that ratio to reach this level of imbalance.
  a. What remedies do we have if someone manipulates it into this imbalanced state? Is it possible to get funds locked forever?
2. Instead of within1, do withinAccuracy
  b. Or you could still do within1 but at the end instead of reverting, if its within tolerance you return it.
3. Raise an issue with a failing test case on balancer repo, let their maths gigabrains solve it for us

Reply:
This is a difficult question, let us take some time to think through before setting on a solution. Because this decision is critical to the design. 

If somehow the pool gets into this state, we can get it out of that state by, the `burn` function, which is why it is important that it has a failsafe to ensure that it always succeeds. It lodges the state out of that death loop. But this is just my conjecture. I don't have a test to prove that this is indeed the case. 

In my searching of curve, saddle, balancer's git repos and discords, no one seems to have hit this situation in production. But in beethovenx on fantom (balancer fork), they have [hit this problem](https://discord.com/channels/885764705526882335/886037144211173386/973957616755691590) before.

So we see that this will happen when a stablecoin depegs, which is a matter of when not if for us. 

That said, I think option 2 is a good middle ground. The inaccuracy is usually so small that it does not affect the calculations in a very material way. We can define a small enough window such ensure that.

Reply:
Okay, let's discuss in person then